storage_controller/src/service.rs

```diff
@@ -8342,8 +8355,8 @@ impl Service {
                             "Skipping migration of {tenant_shard_id} to {node} because secondary isn't ready: {progress:?}"
                         );
 
-                        #[cfg(feature = "testing")]
-                        if progress.heatmap_mtime.is_none() {
+                        if self.config.kick_secondary_downloads && progress.heatmap_mtime.is_none()
```

Comment:
I'd suggest moving the config check into kick_secondary_download(), so that we don't introduce the possibility that someone later calls it without checking this config